freqtrade/configuration/configuration.py

```diff
@@ -75,6 +76,10 @@ def load_from_files(self, files: List[str]) -> Dict[str, Any]:
         # Normalize config
         if 'internals' not in config:
             config['internals'] = {}
+        # TODO: This can be deleted along with removal of deprecated
+        # experimental settings
+        if 'ask_strategy' not in config:
+            config['ask_strategy'] = {}
```

Comment:
Yes and now -- this should be left then, but not in this place and not in this way.

Well, we've got many checks against non-existing sections in the config, like those with `.get()`:
https://github.com/freqtrade/freqtrade/blob/caf415dc97df3cdb607a2ebef18bbc2851b980a8/freqtrade/exchange/exchange.py#L217-L218

Actually,I think this should be done once, for all sections in the config, that are not filled with the settings and do not exist. Somewhere in the Configuration, as a part of the normalization of the config at the early stage of config loading. So that all those checks could be simply eliminated in the code, to simplify the codebase...

But that's not for this PR.

Reply:
well no.

`order_types` and `order_time_in_force` are good examples as to why this won't work.
We cannot normalize these in the configuration loading, because this is done as part of strategy loading.
(they need to be in either Configuration, or in the strategy).

If you normalize them as part of config loading, then StrategyResolver will assume they are set in config and will always ignore what is set as part of the strategy.
For that, they are validated in one place - in the exchange (they are only relevant when you use trading, which requires the exchange to be loaded).

Reply:
Let's go through the global normalization in a separate PR.

This was inserted here because `process_deprecated_setting` copies the value of the deprecated setting into this section, so it must exist.

In strategy resolver we'vr have:
```
        # make sure ask_strategy dict is available
        if 'ask_strategy' not in config:
            config['ask_strategy'] = {}
```

Reply:
Well now THIS is what i was looking for! 

you're right, it's actually done in StrategyResolver. 
Keep as is then 👍